### PR TITLE
ENH: Providing ITKv5 updates for C++11

### PR DESCRIPTION
**Pull request by hjmjohnson**
This pull request incorporates code changes
that happened in parallel with updating the
core ITK code to ITKv5 for C++11 conformity.

### CODE REVIEW

File: Examples/LesionSegmentationCLI.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ class LesionSegmentationCLI : public MetaCommand
 {
 public:
   using PixelType = short;
-  const static unsigned int ImageDimension = 3;
+  constexpr static unsigned int ImageDimension  = 3;
```
Comment (dzenanz) [inline Examples/LesionSegmentationCLI.h]: extra space before the equals sign

File: include/itkRegionCompetitionImageFilter.hxx [hunk 2]
```diff
@@ -259,7 +259,7 @@ RegionCompetitionImageFilter<TInputImage,TOutputImage>
 
   unsigned int neighborhoodSize = bit.Size();
 
-  const OutputImagePixelType backgroundValue = 0;  // no-label value.
+  constexpr OutputImagePixelType backgroundValue  = 0;  // no-label value.
```
Comment (dzenanz) [inline include/itkRegionCompetitionImageFilter.hxx]: extra space before the equals sign

File: include/itkRegionCompetitionImageFilter.hxx [hunk 3]
```diff
@@ -445,7 +445,7 @@ RegionCompetitionImageFilter<TInputImage,TOutputImage>
 
   NeigborOffsetIterator neighborItr = this->m_NeighborBufferOffset.begin();
 
-  const OutputImagePixelType backgroundValue = 0;  // FIXME: replace with NO-Label.
+  constexpr OutputImagePixelType backgroundValue  = 0;  // FIXME: replace with NO-Label.
```
Comment (dzenanz) [inline include/itkRegionCompetitionImageFilter.hxx]: extra space before the equals sign

Comment (hjmjohnson) [reply include/itkRegionCompetitionImageFilter.hxx]: Thanks @dzenanz .  I'm not going to fix those spaces now.  Perhaps in future code cleanups.

**Pull request merged**